hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/HoodieRealtimeRecordReader.java

```diff
@@ -106,29 +102,22 @@ private void readAndCompactLog() throws IOException {
         // TODO(vc): In the future, the reader schema should be updated based on log files & be able to null out fields not present before
         Schema readerSchema = generateProjectionSchema(writerSchema, projectionFields);
 
-        LOG.info(String.format("About to read logs %s for base split %s, projecting cols %s",
+        LOG.info(String.format("About to read compacted logs %s for base split %s, projecting cols %s",
                 split.getDeltaFilePaths(), split.getPath(), projectionFields));
-        for (String logFilePath: split.getDeltaFilePaths()) {
-            GenericDatumReader<GenericRecord> datumReader = new GenericDatumReader<>(writerSchema, readerSchema);
-            final AvroFSInput input = new AvroFSInput(FileContext.getFileContext(jobConf), new Path(logFilePath));
-            DataFileReader<GenericRecord> reader = (DataFileReader<GenericRecord>) DataFileReader.openReader(input, datumReader);
-            while (reader.hasNext()) {
-                GenericRecord rec = reader.next();
-                String key = rec.get(HoodieRecord.RECORD_KEY_METADATA_FIELD).toString();
```

Comment:
Just confirming here, that the iterator on the right won't read past the a known maximum commit time.. There will always be data being appended as we read the log files.